src/emitter.ts

```diff
@@ -323,11 +315,8 @@ export function emitWebIDl(webidl: Browser.WebIdl, flavor: Flavor) {
                     allEnumsMap[objDomType]) return objDomType;
                 // Name of a type alias. Just return itself
                 if (allTypeDefsMap.has(objDomType)) return objDomType;
-                // Union types
-                if (objDomType.includes(" or ")) {
-                    const allTypes: string[] = decomposeTypes(objDomType).map(t => convertDomTypeToTsTypeSimple(t.replace("?", "")));
-                    return allTypes.includes("any") ? "any" : allTypes.join(" | ");
-                }
+                // Known interface name but not for the target flavor
+                if (unfilteredInterfaceNames.has(objDomType)) return "never";
```

Comment:
do not think you need this `unfilteredInterfaceNames` map at all.. you can make `never` the fallback at the end of the function instead of the throw. and in `convertDomTypeToTsType`, add a flag to allow `never` as a type, if the flag is not set, then throw. we only set the flag in function parameters since we are handling `never` there.